users/models/identity.py

```diff
@@ -30,19 +30,71 @@
 
 class IdentityStates(StateGraph):
     """
-    There are only two states in a cycle.
     Identities sit in "updated" for up to system.identity_max_age, and then
     go back to "outdated" for refetching.
+
+    When a local identity is "edited" or "deleted", it will fanout the change to
+    all followers and transition to "updated"
     """
 
     outdated = State(try_interval=3600, force_initial=True)
     updated = State(try_interval=86400 * 7, attempt_immediately=False)
 
+    edited = State(try_interval=300, attempt_immediately=True)
+    deleted = State(externally_progressed=True)
+
+    edited.transitions_to(updated)
+    updated.transitions_to(edited)
+    edited.transitions_to(deleted)
+
     outdated.transitions_to(updated)
     updated.transitions_to(outdated)
 
+    @classmethod
+    async def targets_fan_out(cls, identity: "Identity", type_: str) -> None:
+        from activities.models import FanOut
+        from users.models import Follow
+
+        # Fan out to each target
+        shared_inboxes = set()
+        async for follower in Follow.objects.select_related("source", "target").filter(
+            target=identity
+        ):
+            # Dedupe shared_inbox_uri
+            shared_uri = follower.source.shared_inbox_uri
+            if shared_uri and shared_uri in shared_inboxes:
+                continue
+
+            await FanOut.objects.acreate(
+                identity=follower.source,
+                type=type_,
+                subject_identity=identity,
+            )
+            shared_inboxes.add(shared_uri)
+
+    @classmethod
+    async def handle_new(cls, instance):
+        return cls.outdated
+
+    @classmethod
+    async def handle_edited(cls, instance: "Identity"):
+        from activities.models import FanOut
+
+        if not instance.local:
+            return cls.updated
+
+        identity = await instance.afetch_full()
+
+        if identity.deleted:
```

Comment:
While I did leave a "deleted" field on the model, I'd rather this follows the existing Post pattern where we actually use "deleted" and "deleted_fanned_out" states, and we ignore the "deleted" field on the model entirely (and remove it when the next migration opportunity arises)

Reply:
Changed flow to `deleted` and `deleted_fanned_out`.